src/material/datepicker/calendar-body.scss

```diff
@@ -152,8 +152,13 @@ $mat-calendar-range-end-body-cell-size:
   cursor: default;
 }
 
-.mat-calendar-body-cell-content {
+// Apply this style with additional specificity to override the focus indicator's relative
+// positioning.
+.mat-calendar-body-cell .mat-calendar-body-cell-content {
```

Comment:
nit: you can move this back under `.mat-calendar-body-cell-content` and change the selector to `.mat-calendar-body-cell &`.